scalding-core/src/main/scala/com/twitter/scalding/Execution.scala

```diff
@@ -467,27 +467,60 @@ object Execution {
    * This is here so we can call without knowing the type T
    * but with proof that pipe matches sink
    */
-  private case class ToWrite[T](pipe: TypedPipe[T], sink: TypedSink[T]) {
-    def write(flowDef: FlowDef, mode: Mode): Unit = {
-      // This has the side effect of mutating flowDef
-      pipe.write(sink)(flowDef, mode)
-      ()
+
+  private trait ToWrite[T] {
```

Comment:
don't think we need the `T` here, actually. Your design makes that clearer. Nice.